tests/unit/src/test/scala/tests/testProvider/ScalatestFinderSuite.scala

```diff
@@ -0,0 +1,227 @@
+       |}
+       |""".stripMargin,
+    FullyQualifiedName("SetSpec"),
+    Vector(
+      ("A Set when empty should have size 0", QuickRange(4, 6, 4, 19)),
+      (
+        "A Set when empty should produce NoSuchElementException when head is invoked",
+        QuickRange(8, 6, 8, 59),
+      ),
+    ),
+    ScalatestStyle.AnyWordSpec,
+  )
+
+  check(
+    "any-flat-spec",
+    """|import org.scalatest.flatspec.AnyFlatSpec
+       |
+       |class FlatSpec extends AnyFlatSpec {
+       |
+       |  "An empty Set" should "have size 0" in {
+       |    assert(Set.empty.size == 0)
+       |  }
+       |
+       |  it should "produce NoSuchElementException when head is invoked" in {
+       |    assertThrows[NoSuchElementException] {
+       |      Set.empty.head
+       |    }
+       |  }
+       |}
+       |""".stripMargin,
+    FullyQualifiedName("FlatSpec"),
+    Vector(
+      (
+        "An empty Set should produce NoSuchElementException when head is invoked",
+        QuickRange(8, 2, 8, 65),
+      ),
+      ("An empty Set should have size 0", QuickRange(4, 2, 4, 37)),
+    ),
+    ScalatestStyle.AnyFlatSpec,
+  )
+
+  check(
+    "any-fun-spec",
+    """|import org.scalatest.funspec.AnyFunSpec
+       |
+       |class FunSpec extends AnyFunSpec {
+       |
+       |  describe("A Set") {
+       |    describe("when empty") {
+       |      it("should have size 0") {
+       |        assert(Set.empty.size == 0)
+       |      }
+       |
+       |      it("should produce NoSuchElementException when head is invoked") {
+       |        assertThrows[NoSuchElementException] {
+       |          Set.empty.head
+       |        }
+       |      }
+       |    }
+       |  }
+       |}
+       |""".stripMargin,
+    FullyQualifiedName("FunSpec"),
+    Vector(
+      ("A Set when empty should have size 0", QuickRange(6, 6, 6, 30)),
+      (
+        "A Set when empty should produce NoSuchElementException when head is invoked",
+        QuickRange(10, 6, 10, 70),
+      ),
+    ),
+    ScalatestStyle.AnyFunSpec,
+  )
+
+  check(
+    "any-free-spec",
+    """|import org.scalatest.freespec.AnyFreeSpec
+       |
+       |class FreeSpec extends AnyFreeSpec {
+       |
+       |  "A Set" - {
+       |    "when empty" - {
+       |      "should have size 0" in {
+       |        assert(Set.empty.size == 0)
+       |      }
+       |
+       |      "should produce NoSuchElementException when head is invoked" in {
+       |        assertThrows[NoSuchElementException] {
+       |          Set.empty.head
+       |        }
+       |      }
+       |    }
+       |  }
+       |}
+       |""".stripMargin,
+    FullyQualifiedName("FreeSpec"),
+    Vector(
+      ("A Set when empty should have size 0", QuickRange(6, 6, 6, 26)),
+      (
+        "A Set when empty should produce NoSuchElementException when head is invoked",
+        QuickRange(10, 6, 10, 66),
+      ),
+    ),
+    ScalatestStyle.AnyFreeSpec,
+  )
+
+  check(
+    "any-prop-spec",
```

Comment:
Do we have any tests that actually run the test cases that were found? If not that would be great to add at least some.

Reply:
Such tests would be a great addition, but unfortunately there aren't any at all. Good idea for future